### DISCUSSION TITLE
Prisma keeps query running even after response

### DISCUSSION THREAD

**Discussion by eugenrecaj** [prisma/prisma#20922] [Q&A] [answered]
### Bug description

```prisma
model Post {
  id        Int       @id @default(autoincrement())
  description String
  createdAt   DateTime  @default(now())
  updatedAt   DateTime  @updatedAt

  @@index([id])
}

model User {
  id        Int       @id @default(autoincrement())
  email     String    @unique
  firstname String
  lastname  String
  phone     String?
  password  String?
  bio       String?
  twitter   String?   @default("")
  facebook  String?   @default("")
  instagram String?   @default("")
  snapchat  String?   @default("")
  createdAt   DateTime  @default(now())
  updatedAt   DateTime  @updatedAt

  @@index([id])
  @@index([email])
}
```

I have these 2 models on my prisma.schema and when I try to get all Posts:

```js
const allPosts= await this.prisma.post.findMany();
```

I get the response on client side and everything works fine but the database shows the query running:

[image](https://github.com/prisma/prisma/assets/18050599/caf6c5bd-660d-4cb7-ab5f-a0c1eabe5861)

But when I try to get all users:

```js
const allUsers= await this.prisma.user.findMany();
```
This issue does not happen. There is no running query for users only for posts.

### How to reproduce

Using prisma v5.2.0 with nestjs.

**Comment by janpio** [comment]
That should be impossible, right? When the data is returned, the query has finished already.
Does it count up the "Duration" still?
Or does the UI maybe still show the final query for each connection?
What UI is this?

**Reply by eugenrecaj** [reply]
I was expecting that behavior that after data returns the query ends. But no.
The duration keeps going up and if no timeout is set it runs all the time.
If I create other models and run them, most of them will remain running. I tried to run "allPosts" and "allUsers" one after another:

```
const allPosts= await this.prisma.post.findMany();
const allUsers= await this.prisma.user.findMany();
```

The "allPosts" query does not appear running. But if I switch the positions:

```
const allUsers= await this.prisma.user.findMany();
const allPosts= await this.prisma.post.findMany();
```

Then the "allPosts" query keeps running even after returning all posts.

I am using Postgres DB

This is DigitalOcean databases. I also tried it with other DB providers and had other people replicate it as well.
Everyone had the same problem.

**Reply by eugenrecaj** [reply]
I tried it with MySql DB and it still keeps the query running but this time it shows different and the duration is 0ms

![image](https://github.com/prisma/prisma/assets/18050599/8a93da2f-036a-4620-b027-ae4b52854bdc)

**Reply by janpio** [reply, marked as answer]
If Prisma returns the data, the query has succeeded on the database side and it is finished. Prisma actually only gets the data _after_ it has finished. Prisma also does not "run" a query itself, it just asks the database "please run this query and give me the data" - and the database executes that and then returns the data.

But I can confirm what you are seeing on a DO database. It sometimes reports queries as still running, which definitely already finished.
<img width="933" alt="image" src="https://github.com/prisma/prisma/assets/183673/cc7f21f0-d3c9-4927-9bde-dc51d98fa5e3">

I think that is a problem on the DO side.

(For me this is only happening on databases that use DO's `Connection Pool`. Are you also seeing that there only or also for the direct connection string?)

**Reply by eugenrecaj** [reply]
It's true, this is happening on DO. The reason is unknown for now. Thanks for taking time to look into it.
